engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeConversionNode.java

```diff
@@ -247,7 +250,7 @@ Object doConvertDate(
       Object that,
       Object[] arguments,
       @Shared("interop") @CachedLibrary(limit = "10") InteropLibrary interop,
-      @Shared("typesLib") @CachedLibrary(limit = "10") TypesLibrary typesLib,
+      @Shared("typesLib") @Cached TypeOfNode typesLib,
```

Comment:
```suggestion
      @Shared("typesLib") @Cached TypeOfNode typeOfNode,
```

@JaroslavTulach The type has changed but the name remains as before - shouldn't it be updated?

IMO having names that are inconsistent with what the value is, is likely to introduce confusion.

Reply:
Names of local variables don't have any effect on the execution.

Reply:
> Names of local variables don't have any effect on the execution. 

I know! I'm only complaining about code style. I find such small practices really greatly influencing how easy it is for me to understand the code. The variable name _tells something_. Sometimes it doesn't say much useful information than the type. But in cases like here, when there's inconsistency - I instantly start to think why the name is the way it is. In the diff I know - for historical reasons, because during refactor the variable names were not updated. But if I will come looking at this code later, without all the historical context - I will get confused and will be wondering what does this unusual name mean - why this instance is a `Lib` if its type suggests it's a `Node` - what is happening here? And I will lose time unproductively wondering what has happened here.

I know I'm saying as if this is a big thing. But for me _it is_. Of course a single variable name is not going to cause harm, but if there's a lot of such situations throughout the codebase, I start wasting more and more time trying to understand what the author meant, and processing the codebase when trying to extend it or fix something takes significantly longer when such things add up.

I guess you are familiar with this part of the codebase, so these things are obvious to you. But if I'm less familiar with it, even such minor details do affect how hard it is to understand pieces of the code. Do you really never experience e.g. bad naming causing confusion and causing you to take more time to understand the codebase? 

I once heard a wise saying that _code is read more times than it is written_ - so IMO taking a little bit of extra effort in making it easy to understand greatly pays off in the long term.

I really don't want to be complaining all the time about this, I hope the tone of my complaint above is not too offensive. I just really wanted to share _how very frustrated_ I get when I have a hard time understanding parts of the codebase littered with cryptic variable names or little documentation.

> Your suggestion cannot be applied on its own - it would make the code uncompilable.

Sorry, I used suggestion to show how the rename can be applied, but I know we need to update other occurrences. I felt it's enough to suggest what rename is needed, not the whole patch. I may try adding better suggestions next time, but it's not trivial to find all usages in the GitHub review, whereas with any IDE such rename is usually much easier.

Reply:
@JaroslavTulach this fix worked, thank you very much.
@radeusgd I'll do the renaming.